core/include/detray/geometry/detail/propagation_kernels.hpp

```diff
@@ -0,0 +1,132 @@
+/** Detray library, part of the ACTS project (R&D line)
```

Comment:
I don't think I can compile geometry-only if I have to include the propagation focused kernels in the ```surface_kernels.hpp```? Also, how will the ```operator()``` overload work between e.g. ```free_to_bound_vector``` and ```free_to_bound_jacobian``` if I put them in the same struct, since they take the same arguments? Or maybe I don't understand what you mean :) I can put the structs into the ```propagation``` namespace and but the typedefs in there maybe? Otherwise, I would not like to have the typedefs globally

Reply:
Sth like this

```c++
template <typename algebra_t>
class surface_kernels{
  
    using transform3 = algebra_t;
    using free_vector_type =
        typename free_track_parameters<algebra_t>::vector_type;
    using bound_vector_type =
        typename bound_track_parameters<algebra_t>::vector_type;

   struct free_to_bound_vector{
           template <typename mask_group_t, typename index_t>
      //...
    
   };
   
   struct free_to_bound_jacobian{
     //...
   };
}
```

Reply:
Ah, nvm, the ```algebra``` template...

Reply:
You can choose whatever you prefer. It was totally my personal preference and the current state is also OK to me